src/components/warning-text/template.test.js

```diff
@@ -0,0 +1,78 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('warning-text')
+
+describe('Warning text', () => {
+  it('renders the default example with text and assistive text', () => {
+    const { $ } = render('warning-text', examples.default)
+
+    const $component = $('.govuk-c-warning-text')
+    expect($component.text()).toContain('You can be fined up to £5,000 if you don’t register.')
+
+    const $assistiveText = $('.govuk-c-warning-text__assistive')
+    expect($assistiveText.text()).toEqual('Warning')
+  })
+
+  it('renders the default example with aria attribute for icon', () => {
+    const { $ } = render('warning-text', examples.default)
+
+    const $icon = $('.govuk-c-warning-text__icon')
+    expect($icon.attr('aria-hidden')).toEqual('true')
+  })
+
+  it('renders classes', () => {
+    const { $ } = render('warning-text', {
+      classes: 'govuk-c-warning-text--custom-class',
+      text: 'Warning text'
+    })
+
+    const $component = $('.govuk-c-warning-text')
+    expect($component.hasClass('govuk-c-warning-text--custom-class')).toBeTruthy()
+  })
+
+  it('renders custom text and custom assistive text', () => {
```

Comment:
I think this would make more sense as two different tests.

Reply:
I was in two minds about that but I think you're right. Have split it into two tests.